siliconcompiler/core.py

```diff
@@ -2751,14 +2751,22 @@ def check_logfile(self, jobname=None, step=None, index='0',
             checks[suffix]['args'] = regexes
             matches[suffix] = 0
 
+        # Order suffixes as follows: [..., 'warnings', 'errors']
+        ordered_suffixes = list(filter(lambda key:
+                                       key not in ['warnings', 'errors'], checks.keys()))
+        if 'warnings' in checks:
+            ordered_suffixes.append('warnings')
+        if 'errors' in checks:
+            ordered_suffixes.append('errors')
+
         # Looping through patterns for each line
         with open(logfile, errors='ignore_with_warning') as f:
             line_count = sum(1 for _ in f)
             right_align = len(str(line_count))
-            # Start at the beginning of file again
-            f.seek(0)
-            for num, line in enumerate(f, start=1):
-                for suffix in checks:
+            for suffix in ordered_suffixes:
+                # Start at the beginning of file again
+                f.seek(0)
+                for num, line in enumerate(f, start=1):
```

Comment:
Can you make it such that warning lines print before the error lines.

Reply:
That's already the case as is.
I adjusted the test now to make it more obvious.
We now have an error in the tool log before a warning.
In the siliconcompiler log the warning comes before the error.